lib/reply.js

```diff
@@ -298,18 +298,18 @@ function handleError (reply, error, cb) {
 
   res.statusCode = statusCode
 
-  if (statusCode >= 500) {
-    res.log.error({ req: reply.request.raw, res: res, err: error }, error && error.message)
-  } else if (statusCode >= 400) {
-    res.log.info({ res: res, err: error }, error && error.message)
-  }
-
-  var customErrorHandler = reply.context.errorHandler
-  if (customErrorHandler && reply._customError === false) {
+  var errorHandler = reply.context.errorHandler
+  if (!errorHandler) {
+    if (statusCode >= 500) {
+      res.log.error({ req: reply.request.raw, res: res, err: error }, error && error.message)
+    } else if (statusCode >= 400) {
+      res.log.info({ res: res, err: error }, error && error.message)
+    }
```

Comment:
I think we should _always_ have an `errorHandler` that we call.

Reply:
I initially though of adding it in Context if errorHandler was missing
```js
this.errorHandler = errorHandler || defaultErrorHandler
```
But it was breaking some tests

Reply:
what was that breaking?

Reply:
```
test/stream.test.js
test/http2/http2.test.js
test/internals/handleRequest.test.js
```

To replicate:
```js
  function defaultErrorHandler (error, request, reply) {
    var res = reply.res
    var statusCode = res.statusCode ? res.statusCode : 500
    if (statusCode >= 500) {
      res.log.error({ req: reply.request.raw, res: res, err: error }, error && error.message)
    } else if (statusCode >= 400) {
      res.log.info({ res: res, err: error }, error && error.message)
    }
    reply.send(error)
  }
```
```js
function Context(...) {
  ...
  this.errorHandler = errorHandler || defaultErrorHandler
  ...
}
```

Reply:
That is weird, we'll have to investigate. `test/internals/handleRequest.test.js` would need to be adjusted because we mock a lot of things there (so that _should_ fail, and would need to be updated). How are the other failing? Can you paste?

Reply:
Were you able to have a look? I pushed a failing branch if you need https://github.com/mixecan/fastify/tree/failing-default-handler